src/services/actions/userActions.ts

```diff
@@ -32,3 +35,103 @@ export async function makeRegularUser(user: { userId: string }) {
     revalidatePath('/', 'layout')
   }
 }
+
+export async function updateUserProfile(
+  data: ProfileDataValues,
+) {
+  const validatedData = profileDataSchema.parse(data)
+   try {
+    const {id,picture, name, phoneNumber, stopCommunication,email,...socialData } = validatedData
+    return await prisma.$transaction(async (tx) => {
+      const currentUser = await tx.user.findUnique({
+        where: { id: id },
+        select: { dataCollected: true }
+      });
+
+      const updatedUser = await tx.user.update({
+        where: { id: id },
+        data: {
+          name,
+          phoneNumber,
+          stopCommunication,
+          picture,
+          email,
+          dataCollected: {
+            ...(currentUser?.dataCollected as Record<string, unknown>),
+            ...socialData
+          }
+        }
+      });
+      revalidatePath('/user/edit');
+      return { success: true, user: updatedUser };
+    });
+  } catch (error) {
+    console.error('Error updating profile:', error);
+    throw new Error('Failed to update profile');
+  }
+}
+export async function getUserProfile() {
+  try {
+    const userCurrent = await getUser()
+    if (!userCurrent?.id) return null
+    const user = await prisma.user.findUnique({
+      where: { id: userCurrent.id },
+      select: {
+        id: true,
+        name: true,
+        email: true,
+        picture: true,
+        phoneNumber: true,
+        stopCommunication: true,
+        dataCollected: true
+      }
+    })
+    if (!user) return null 
+    const dataCollected = user.dataCollected as UserDataCollected
+    const cleanUserData = {
```

Comment:
you can just pass the object directly to parse. no need to do this. u can provide zod with a default value